pennylane/spin/lattice.py

```diff
@@ -0,0 +1,196 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains functions and classes to create a
+:class:`~pennylane.spin.lattice` object. This object stores all
+the necessary information about a lattice.
+"""
+import itertools
+
+import numpy as np
+from scipy.spatial import cKDTree
+
+# pylint: disable=too-many-arguments, too-many-instance-attributes
+# pylint: disable=use-a-generator, too-few-public-methods
+
+
+class Lattice:
+    r"""Constructs a Lattice object.
+
+    Args:
+       L: Number of unit cells in each direction.
+       unit_cell: Primitive vectors for the lattice.
+       basis: Initial positions of spins.
+       boundary_condition: defines boundary conditions, boolean or series of bools with dimensions same as L.
```

Comment:
I realized skipping `periodic` word must have come from a larger discussion - but this description doesn't tell me what does defining the values of boundary conditions would do. Might consider rewording.

Reply:
Also the starting letter should be uniform for all.